test/6_Events.js

```diff
@@ -0,0 +1,152 @@
+
+      let date = new Date();
+      date.setDate(date.getDate() + 1);
+
+      const future = Math.floor(date.getTime() / 1000);
+      const owner = accounts[1];
+      const tx = await lockedDeal.CreateNewPool(testToken.address, future, allow, owner, { from: fromAddress });
+      const poolId = tx.logs[1].args.PoolId;
+
+      result = tx;
+    });
+
+    it('NewPoolCreated event is emitted', async () => {
+      // Check event
+      truffleAssert.eventEmitted(result, 'NewPoolCreated');
+    });
+
+  });
+
+  describe('PoolOwnershipTransfered event is emitted', async () => {
+
+    before(async () => {
+      await testToken.approve(lockedDeal.address, allow, { from: fromAddress });
+
+      let date = new Date();
+      date.setDate(date.getDate() + 1);
+
+      let future = Math.floor(date.getTime() / 1000);
+      await lockedDeal.CreateNewPool(testToken.address, future, allow, owner, { from: fromAddress });
+      // poolId = tx.logs[1].args.PoolId
+
+      await testToken.approve(lockedDeal.address, allow, { from: fromAddress });
+      // let date = new Date()
+      date.setDate(date.getDate() + 1);
+      future = Math.floor(date.getTime() / 1000);
+      const tx = await lockedDeal.CreateNewPool(testToken.address, future, allow, owner, { from: fromAddress });
+      poolId = tx.logs[1].args.PoolId;
+    });
+
+    it('PoolOwnershipTransfered event is emitted', async () => {
+      const newOwner = accounts[8];
+
+      result = await lockedDeal.TransferPoolOwnership(poolId, newOwner, { from: owner });
+      // Check event
+      truffleAssert.eventEmitted(result, 'PoolOwnershipTransfered');
+    });
+    
+  });
+
+  describe('PoolApproval event is emitted', async () => {
+
+    const approvalAmount = 10;
+    const spender = accounts[1];
+
+    before(async () => {
+      await testToken.approve(lockedDeal.address, allow, { from: fromAddress });
+
+      let date = new Date();
+      date.setDate(date.getDate() + 1);
+
+      let future = Math.floor(date.getTime() / 1000);
+      const tx = await lockedDeal.CreateNewPool(testToken.address, future, allow, owner, { from: fromAddress });
+      poolId = tx.logs[1].args.PoolId;
+    });
+
+    it('PoolApproval event is emitted', async () => {
+      result = await lockedDeal.ApproveAllowance(poolId, approvalAmount, spender, { from: owner });
+      // Check event
+      truffleAssert.eventEmitted(result, 'PoolApproval');
+    });
+    
+  });
+
+  describe('PoolSplit event is emitted', async () => {
+    const amount = 10;
+
+    before(async () => {
+      allow -= amount;
+
+      await testToken.approve(lockedDeal.address, allow, { from: fromAddress });
+
+      let date = new Date();
+      date.setDate(date.getDate() + 1);
+
+      let future = Math.floor(date.getTime() / 1000);
+
+      const tx = await lockedDeal.CreateNewPool(testToken.address, future, allow, owner, { from: fromAddress });
+      poolId = tx.logs[1].args.PoolId;
```

Comment:
I'm not sure about this test. can you use truffleAssert.eventEmitted

Reply:
It is not a full test, you can see below all lines of test. I used truffleAssert.eventEmitted.
That lines upper are initial prepare for the test.
![image](https://user-images.githubusercontent.com/45734486/151168006-2d43e0c9-5da3-46a4-af3a-6762beeb900b.png)